### DISCUSSION TITLE
wildcard console

### DISCUSSION THREAD

**Discussion by kingspride** [bstansell/conserver#104] [General]
Hi,
I'd like to know if its possible to tier conservers?
I use a master conserver which often just refers to another host that actually holds the console (mostly unix socket, thats why)
this is what the config looks like:
```
# The character '&' in logfile names are substituted with the console
# # name.
# #
#
 config * {
	autocomplete on;
 	sslrequired no;
	reinitcheck 5;
}

default full {
	rw *;
}
default * {
	logfile /var/log/conserver/&.log;
	timestamp "";
	include full;
	#sslrequired no;
}
#
default softioc { type host; host localhost;}
default softiocunix { type exec; }
#include /etc/conserver/iocs.cf

##
## list of clients we allow
##
access * {
	trusted 127.0.0.1;
	trusted 10.0.0.0/23;
}
```

`iocs.cf` then holds the console information:
```
[...]
console iocTEST {include softiocunix; master ioc-host-1;}
[...]
```
Connecting to iocTEST then "just works" by `console iocTEST`.

I noticed that when issueing `console -x` or `console -i`, I get info about all consoles the lower tiered conservers hold.
but, I cant connect to them via `console myconsolename` if the console is not in the config of the master conserver.

would it be somehow possible to tell the master conserver subsidary conservers?

That would solve my issue where I always have to configure new consoles on the master conserver after setting them up on a host. One tends to forget and then users cant connect to the console...

thanks in advance,
William

PS: I hope this project is not dead yet :)

**Comment by robohack** [comment]
There exists somewhat buggy, and undocumented, support for declaring a list of hosts as the `master` for the `console` client, so I just have a `/etc/console.cf` that lists all the hosts running conserver (prefixed by `localhost` to work around bugs):

    config * {
            master localhost:782@xentastic.local:782@xenful.local;
    }

Here `xentastic` and `xenful` are two Xen servers which run `conserver` to allow connections to their domUs (VMs).

The result is I can see and connect to any of the remote consoles as well as those defined for the local `master` conserver instance:

```
# console -x
 xentastic                on 10.0.1.240/2005                  at   Netwk 
 isit                     on 10.0.1.240/2004                  at   Netwk 
 xenful                   on 10.0.1.240/2003                  at   Netwk 
 once                     on 10.0.1.240/2002                  at   Netwk 
 xentastic                on 10.0.1.240/2005                  at   Netwk 
 isit                     on 10.0.1.240/2004                  at   Netwk 
 xenful                   on 10.0.1.240/2003                  at   Netwk 
 once                     on 10.0.1.240/2002                  at   Netwk 
console: Unable to connect to 127.0.0.1:1
 fezzik                   on /dev/pts/1@localhost             at 115200n
 nbtest                   on /dev/pts/0@localhost             at 115200n
 dummy-xentastic          on /dev/pts/5                       at  Local 
 nbtcur                   on /dev/pts/1@localhost             at 115200n
 more                     on /dev/pts/0@localhost             at 115200n
 dummy-xenful             on (null)                           at  Local 
```

Note the `dummy-*` consoles are there because conserver insists on there always being one `console` entry, e.g. if there are no VMs running.

I manage the console lists on the Xen servers with some helper scripts:

https://github.com/robohack/mk-conserver-xen

**Reply by kingspride** [reply]
Thanks for your input. I'll try that and see if it does what I want :)

edit: I also have dummy consoles for the exact same reason..

**Comment by kingspride** [comment]
okay, little follow up ...

I tried what you described here, and it works. :)

But, I just realized, this only really forwards my problem, to configure all conservers somewhere, to _all_ clients :(
thats unfortunately worse than now, where I just have to maintain an iocs.cf file on my super-master-conserver where all my consoles are documented.

would have been nice to have a config like
```
console * { master ioc-host-1.local; }
console * { master ioc-host-2.local; }
[...]
```
and after that, clients would get redirected to the corresponding host.

I'll see if I get another idea how to handle this,
in the meantime - thanks for this great tool, I hope it wont die... 
also @robohack thanks again for your idea!

- W
